content/influxdb/v2.0/write-data/troubleshoot.md

```diff
@@ -0,0 +1,74 @@
+weight: 106 >
+  Troubleshoot issues writing data. Discover how writes fail, including rate limit failures, timeouts, size of write payload, not conforming to an explicit schema bucket, and partial writes. Find response codes for failed writes, and suggestions to fix. 
+menu:
+  influxdb_2_0:
+    name: Troubleshoot issues
+    parent: Write data
+influxdb/v2.0/tags: [write, line protocol, errors]
+related:
+  - /influxdb/v2.0/api/#tag/Write, InfluxDB API /write endpoint
+  - /influxdb/v2.0/reference/internals
+  - /influxdb/v2.0/reference/cli/influx/write
+---
+Learn how to handle and recover from errors when writing to InfluxDB.
+
+- [Discover common failure scenarios](#common-failure-scenarios)
+- [HTTP status codes](#http-status-codes)
+- [Troubleshoot failures](#troubleshoot-failures)
+
+## Common failure scenarios
+
+Write requests made to InfluxDB may fail for a number of reasons.
+Common failure scenarios that return an HTTP `4xx` or `5xx` error status code include the following:
+
+- The request exceeded a rate limit.
+- The API token was invalid.
+- The client or server reached a timeout threshold.
+- The size of the data payload was too large.
+- The data was not formatted correctly.
+
+Writes may fail partially or completely even though InfluxDB returns an HTTP `2xx` status code for a valid request. For example, a partial write may occur when InfluxDB writes all points that conform to the bucket schema, but rejects points that have the wrong data type in a field.
+
+## HTTP status codes
+
+InfluxDB uses conventional HTTP status codes to indicate the success or failure of a request.
+Write requests return the following status codes:
+
+- `204` Success. The sent data was correctly formatted and accepted for writing to the bucket. `204` doesn't indicate a successful write operation since writes are asynchronous. See how to [check for rejected points](#review-rejected-points).
+- `400` Bad request. The error indicates the line protocol payload was malformed. The response body contains the first malformed line in the data. All payload data was rejected and not written.
+- `401` Unauthorized. The error may indicate one of the following:
+  - The Authorization: Token header is missing or malformed.
+  - The API token value is missing from the header.
+  - The token does not have sufficient permissions to write to this organization and bucket.
+- `404` Not found. A requested resource was not found. The response body contains the requested resource type, e.g. organization name or bucket, and name.
+- `413` Request entity too large. All payload data was rejected and not written.
+- `429` Too many requests. The token is temporarily over quota. The Retry-After header describes when to try the write again.
+- `500` Internal server error. The default HTTP status for an error.
+- `503` Service unavailable. Server is temporarily unavailable to accept writes. The Retry-After header describes when to try the write again.
```

Comment:
```suggestion
- `204` **Success**: Data was correctly formatted and accepted for writing to the bucket.

    {{% note %}}
`204` doesn't indicate a successful write operation since writes are asynchronous. See how to [check for rejected points](#review-rejected-points).
    {{% /note %}}

- `400` **Bad request**: Indicates the line protocol payload was malformed. The response body contains the first malformed line in the data. All payload data was rejected and not written.
- `401` **Unauthorized**: May indicate one of the following:
  - The `Authorization: Token` header is missing or malformed.
  - The API token value is missing from the header.
  - The API token does not have sufficient permissions to write to the organization and bucket.
- `404` **Not found**: A requested resource was not found. The response body contains the requested resource type, e.g. organization name or bucket, and name.
- `413` **Request entity too large**: All payload data was rejected and not written.
- `429` **Too many requests**: API token is temporarily over quota. The `Retry-After` header describes when to try the write request again.
- `500` **Internal server error**: Default HTTP status for an error.
- `503` **Service unavailable**: Server is temporarily unavailable to accept writes. The `Retry-After` header describes when to try the write again.
```

Will OSS ever return a 429? Are write request limits configurable/enforcable in OSS? The only one that I'm aware of is `http-write-timeout`.